official/transformer/transformer_main.py

```diff
@@ -249,141 +265,185 @@ def train_schedule(
 
     # Train the model for single_iteration_train_steps or until the input fn
     # runs out of examples (if single_iteration_train_steps is None).
-    estimator.train(dataset.train_input_fn, steps=single_iteration_train_steps)
+    estimator.train(
+        dataset.train_input_fn, steps=single_iteration_train_steps,
+        hooks=train_hooks)
 
     eval_results = estimator.evaluate(dataset.eval_input_fn)
     print("Evaluation results (iter %d/%d):" % (i + 1, train_eval_iterations),
           eval_results)
+    benchmark_logger.log_evaluation_result(eval_results)
 
     if evaluate_bleu:
-      uncased_score, _ = evaluate_and_log_bleu(
-          estimator, bleu_writer, bleu_source, bleu_ref)
-      if bleu_threshold is not None and uncased_score > bleu_threshold:
+      uncased_score, cased_score = evaluate_and_log_bleu(
+          estimator, bleu_source, bleu_ref, vocab_file_path)
+
+      # Write bleu scores using summary writer and benchmark logger
+      global_step = get_global_step(estimator)
+      summary = tf.Summary(value=[
+          tf.Summary.Value(tag="bleu/uncased", simple_value=uncased_score),
+          tf.Summary.Value(tag="bleu/cased", simple_value=cased_score),
+      ])
+      bleu_writer.add_summary(summary, global_step)
+      bleu_writer.flush()
+      benchmark_logger.log_metric(
```

Comment:
What's the difference between the eval_results and the value here?

Reply:
The results from estimator.evaluate() are based on an approximate translation (long story short, approx. translations are heavily based on the target golden values provided).
The function `evaluate_and_log_bleu` uses the estimator.predict() path to compute the translations, where the golden values are not provided. The translations are compared to reference file to get the actual bleu score.

Reply:
Can this be added in a comment?